src/api/services/participantsService.ts

```diff
@@ -75,3 +81,44 @@ export const deleteSharingParticipants = async (
   );
   return fetchSharingParticipants(response);
 };
+
+const idParser = z.object({
+  participantId: z.coerce.number(),
+});
+
+export const hasParticipantAccess = async (
+  req: ParticipantRequest,
+  res: Response,
+  next: NextFunction
+) => {
+  const { participantId } = idParser.parse(req.params);
+  const participant = await Participant.query().findById(participantId);
+  if (!participant) {
+    return res.status(404).send([{ message: 'The participant cannot be found.' }]);
+  }
+
+  if (!(await isUserBelongsToParticipant(req.auth?.payload?.email as string, participantId))) {
+    return res.status(403).send([{ message: 'You do not have permission to update participant.' }]);
```

Comment:
Minor: Now that this is an enricher, the 403 should probably not include the word 'update' because it might just be a read. So maybe change to "You do not have permission to that participant."